File: ulyp-ui/src/main/java/com/ulyp/ui/util/StyledText.java
package com.ulyp.ui.util;

import javafx.scene.text.Text;

public class StyledText {

    public static Text of(String content, CssClass style) {
        Text text = new Text(content);
        text.getStyleClass().addAll(style.getCssClasses());
        return text;
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/util/StyledText.kt
package com.ulyp.ui.util

import javafx.scene.text.Text

object StyledText {

    @JvmStatic
    fun of(content: String?, style: Style): Text {
        val text = Text(content)
        text.styleClass.addAll(style.cssClasses)
        return text
    }

    @JvmStatic
    fun of(text: Text, style: Style): Text {
        text.styleClass.addAll(style.cssClasses)
        return text
    }

    @JvmStatic
    fun of(text: Text, vararg styles: Style): Text {
        styles.forEach { text.styleClass.addAll(it.cssClasses) }
        return text
    }

    @JvmStatic
    fun of(content: String?, vararg styles: Style): Text {
        val text = Text(content)
        styles.forEach { text.styleClass.addAll(it.cssClasses) }
        return text
    }
}